file_writer.go

```diff
@@ -77,11 +82,21 @@ func (c *Client) CreateFile(name string, replication int, blockSize int64, perm
 		return nil, &os.PathError{"create", name, interpretCreateException(err)}
 	}
 
+	var enc *transparentEncryptionInfo
+	if createResp.GetFs().GetFileEncryptionInfo() != nil {
+		enc, err = c.kmsGetKey(createResp.GetFs().GetFileEncryptionInfo())
+		if err != nil {
+			c.Remove(name)
```

Comment:
This doesn't strike me as 100% right (especially ignoring the error). Maybe we can get the key before creating the file? What does the java version do?

Reply:
as i understand, java version does not remove file:

after [dfs.create](https://github.com/naver/hadoop/blob/master/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/DistributedFileSystem.java#L404=), they wrap stream with [createWrappedOutputStream](https://github.com/naver/hadoop/blob/9e1123d0a983297d48ec53f786731e52c498d6b4/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/DFSClient.java#L1489=), and i see no `catch`'s to handle error.